activation/activation.go

```diff
@@ -244,34 +249,47 @@ func (b *Builder) buildNipstChallenge(epoch types.EpochId) error {
 func (b *Builder) StartPost(rewardAddress address.Address, dataDir string, space uint64) error {
 	b.log.Info("Starting post, reward address: %x", rewardAddress)
 	b.SetCoinbaseAccount(rewardAddress)
-	b.postInitLock.Lock()
-	if b.initStatus == Done {
-		return nil
-	}
-	if b.initStatus == InProgress {
-		b.postInitLock.Unlock()
-		return fmt.Errorf("attempted to start post when post already initiated")
+
+	initStatus := atomic.LoadInt32(&b.initStatus)
```

Comment:
this does not guarantee we will enter start post only once.

Reply:
Do you refer to the lines before that, or in general?

Reply:
See new implementation.